File: 0890-lemonade-change/0890-lemonade-change.kt
class Solution {
    fun lemonadeChange(bills: IntArray): Boolean {
        var fBills : Int = 0;
        var tBills : Int = 0;

        for (b in bills) {
            if (b == 5) fBills++;
            else if (b == 10) {
                if (fBills <= 0) return false;
                fBills--;
                tBills++;
            } else {
                if (fBills > 0 && tBills > 0) {
                    fBills--;
                    tBills--;
                } else if (fBills >= 3) fBills -= 3;
                else return false;
            }
        }

        return true;
    }
}

File: 0890-lemonade-change/0890-lemonade-change.c
bool lemonadeChange(int* bills, int billsSize) {
    int f = 0, t = 0;

    while (billsSize--) {
        if (*bills == 5) f++;
        else if (*bills == 10) {
            if (f <= 0) return 0;
            f--;
            t++;
        } else {
            if (f > 0 && t > 0) {
                f--;
                t--;
            } else if (f >= 3) f -= 3;
            else return 0;
        }
        bills++;
    }

    return 1;
}